hangb.rb

```diff
@@ -0,0 +1,262 @@
+#    "2nd -  let's start with the whole ship. As long as the user guesses right, the ship stays whole."
+#    "3rd -  when they guess wrong though, take a part of the ship away, and turn it the color
+#            listed after the dot."
+# U: gotcha.
+# m: "Also, about the three quotes. I'm using that to ask you to keep the ship drawing together, double
+# quotes around single quotes. Also, I put an additional backslash in front of backslashes in the
+# ASC II drawing. This "escapes" them, so the drawing displays correctly and doesn't skew."
+#
+  def draw_board
+    if @wrong_count == 0
+      puts """
+    ___________________          _-_
+    \\==============_=_/ ____.---'---`---.____
+                \\_ \\    \\----._________.----/
+                  \\ \\   /  /    `-_-'
+              __,--`.`-'..'-_
+             /____          ||
+                  `--.____,-'
+
+"""
+    elsif @wrong_count == 1
+      puts """
+                                  _-_
+                         ____.---'---`---.____
+                \\_ \\    \\----._________.----/
+                  \\ \\   /  /    `-_-'
+              __,--`.`-'..'-_
+             /____          ||
+                  `--.____,-'
+
+""".cyan
+    elsif @wrong_count == 2
+      puts """
+                                  _-_
+                         ____.---'---`---.____
+                    \\    \\----._________.----/
+                     \\   /  /    `-_-'
+              __,--`.`-'..'-_
+             /____          ||
+                  `--.____,-'
+
+      """.blue
+    elsif @wrong_count == 3
+      puts """
+                                  _-_
+                         ____.---'---`---.____
+                          \\----._________.----/
+                          /  /    `-_-'
+              __,--`.`-'..'-_
+             /____          ||
+                  `--.____,-'
+
+      """.yellow
+    elsif @wrong_count == 4
+      puts """
+                                  _-_
+                         ____.---'---`---.____
+                          \\----._________.----/
+                          /  /    `-_-'
+
+
+
+
+
+      """.magenta
+    elsif @wrong_count == 5
+      puts """
+                                  _-_
+                         ____.---'---`---.____
+                          \\----._________.----/
+                                  `-_-'
+
+
+
+
+
+      """.red
+    else
+      puts "Sorry,WARP CORE BREACH! You lose!".red.blink
+      exit
+    end
+
+  end
+
+# m: " Here's the method definition for @answer_line - It is equal to an empty array.
+#    When you shuffle.pop the "answer" out, create a line of empty underscores exactly the same
+#    length as the "answer." Put all the underscores in the @answer_line array together as a string
+#    - that's what ".join" means. "return" that after you create it, so I (and the user) can see it.
+# U: Word.
+
+  def answer_line
+    @answer_line = []
+    (@answer.length).times do |ans|
```

Comment:
Ruby arrays have a shortcut of creating arrays of a certain size with default value by using its initializer. Check it out:
```
Array.new(@answer.length, ' _ ')
```